### ISSUE TITLE
[6.0.0-rc.4] Problem validate when change form's field

### ISSUE THREAD

**Issue by ViZhe** [erikras/redux-form#1528]
I'm not sure this is a bug or incorrect implementation. Example repository: https://github.com/ViZhe/redux-form-issue
For start:
```
npm i
npm start
```
Open http://0.0.0.0:3000

**The steps describing the problem**
1) I click "Send" button
2) Form show me 2 errors and do action "SET_SUBMIT_FAILED" _- OK_
3) I fix errors in form and submit form again
4) Form do action "SET_SUBMIT_SUCCEEDED" _- OK_
5) Now i can refresh the page or not
6) I select "second" template
7) In form changing some fields
8) I submit this form
9) Form do action "SET_SUBMIT_SUCCEEDED" _- BAD, because ignore validate for field "Size", but the value of this field is passed to the form handler_
10) I touch or write some thing in field "Size"
11) Form show me error "Size must be small"
12) I submit this form
13) Form do action "SET_SUBMIT_SUCCEEDED" _- BAD, because we still ignore validate for field "Size"_

**Why is this happening**
Form does not register new fields and not unregister old.

**UPD:** 
if you have not added new fields and removed some of the old, then:
first, there is the reregistration of the all fields, and then unregistered the all old, and because there is a crossing of the names, unregistered new too.

If a new field was added, then:
Unregistered unused field, and the new is not registered.

Sorry for spelling. Let me know if you need anything.

**Comment by jervi** [comment]
I think this is resolved by the yet unreleased pull request #1486.

**Comment by ViZhe** [comment]
@jervi, Thank for you time.
My steps:
```
git clone git@github.com:erikras/redux-form.git -b v6
cd redux-form
npm i
npm run build
rm -rf node_modules
 ```
Then moved this folder in `project/node_modules`, but result remain the same.

**Comment by erikras** [comment]
Fixed by #1561. I'll leave another comment here when it is released.

**Comment by erikras** [comment]
Fix published in [`v6.0.0-rc.5`](https://github.com/erikras/redux-form/releases/tag/v6.0.0-rc.5).

**Comment by ViZhe** [comment]
@erikras Thank for this.
Confirm previous version of the [repository](https://github.com/ViZhe/redux-form-issue) to start working, but today I [redid](https://github.com/ViZhe/redux-form-issue/commit/95ff720db9d0f4e568d3b460b87f6d9ac1147c7e) it and everything is broken again ;(

**Following in the steps:**
1) When entering the form loads the template "A", resulting in `INITIALIZE` and `REGISTER_FIELD` for fields: "f1", "f2". These fields don't have validation, so when the submit event, I get `SET_SUBMIT_SUCCEEDED` - OK
2) I change the template to "B"
3) Form do `REGISTER_FIELD` for "f3", "f4" and "f5". "f3" is required, but on submit, i get `SET_SUBMIT_SUCCEEDED` - BAD, _because ignore validate for field "f3_
4) I change the template to "C"
5) The following actions occur:
- `INITIALIZE`
- `UNREGISTER_FIELD` for f1
- `REGISTER_FIELD` for f3
- `UNREGISTER_FIELD` for f2
- `REGISTER_FIELD` for f5
- `UNREGISTER_FIELD` for f3
- `REGISTER_FIELD` for f6
- `UNREGISTER_FIELD` for f4
- `UNREGISTER_FIELD` for f5
- `UPDATE_SYNC_ERRORS`
